src/ApparentHorizons/HorizonManagerComponentActions.hpp

```diff
@@ -0,0 +1,652 @@
+                  "### Node:%d  Proc:%d ###\n"
+                  "{%s,%d}: interpolate_data: interped %d pts, now have %d\n\n",
+                  Parallel::my_node(), Parallel::my_proc(), AhTag::label(),
+                  timestep, element_coord_holder.offsets.size(), new_npts);
+            }
+
+            // Here we do more copying than I would like, because
+            // there is no good way to do an "insert" on a Variables.
+            typename VarsTags<typename AhTag::frame>::type new_vars(new_npts);
+            {
+              const auto& old_data = interp_info.vars;
+              for (size_t v = 0; v < n_vars; ++v) {
+                // replace with std::memcpy?
+                for (size_t s = 0; s < old_npts; ++s) {
+                  // clang-tidy: no pointer arithmetic
+                  new_vars.data()[s + v * new_npts] =     // NOLINT
+                      old_data.data()[s + v * old_npts];  // NOLINT
+                }
+              }
+            }
+            for (size_t v = 0; v < n_vars; ++v) {
+              // replace with std::memcpy?
+              for (size_t s = 0; s < delta_npts; ++s) {
+                // clang-tidy: no pointer arithmetic
+                new_vars.data()[s + old_npts + v * new_npts] =  // NOLINT
+                    interp_data.data()[s + v * delta_npts];     // NOLINT
+              }
+            }
+            interp_info.vars = new_vars;
+          }
+        }
+      },
+      make_not_null(&box));
+}
+
+// Sends interpolated data to a ah::Finder.
+template <typename Metavariables, typename DbTags, typename AhTag>
+void reduce_data(db::DataBox<DbTags>& box,
+                 Parallel::ConstGlobalCache<Metavariables>& cache,
+                 const size_t timestep) noexcept {
+  const auto& verbose = db::get<::Tags::Verbosity>(box);
+  auto& holders = db::get<Tags::horizon_holders<Metavariables>>(box);
+  auto& interp_info =
+      get<HorizonHolderTag<AhTag>>(holders).interpolated_vars_info.at(timestep);
+
+  db::mutate<Tags::horizon_holders<Metavariables>>(
+      make_not_null(&box),
+      [&timestep](const gsl::not_null<
+                  db::item_type<Tags::horizon_holders<Metavariables>>*>
+                      holders_l) {
+        auto& interp_info_l = get<HorizonHolderTag<AhTag>>(*holders_l)
+                                  .interpolated_vars_info.at(timestep);
+        interp_info_l.data_has_been_sent = true;
+      });
+
+  if (verbose == ::Verbosity::Debug) {
+    Parallel::printf(
+        "### Node:%d  Proc:%d ###\n"
+        "{%s,%d}: reduce_data: have %d points to send\n\n",
+        Parallel::my_node(), Parallel::my_proc(), AhTag::label(), timestep,
+        interp_info.global_offsets.size());
+  }
+
+  if (not interp_info.global_offsets.empty()) {
+    auto& receiver_proxy =
+        Parallel::get_parallel_component<ah::Finder<Metavariables, AhTag>>(
+            cache);
+    Parallel::simple_action<Actions::Finder::ReceiveInterpolatedVars<AhTag>>(
```

Comment:
check if clean up is done in the function calling this

Reply:
Cleanup of volume data is done after all ah::Finders at that timestep have converged (so none of them need volume data).  Cleanup of interpolated data is done after a particular ah::Finder has converged, and when interpolated data at a new set of points is requested.  Some of the interpolated data could be cleaned up immediately after it is sent.

Reply:
Now cleaning up earlier.